modules/effects/arrow-effects-data/src/main/kotlin/arrow/effects/extensions/io.kt

```diff
@@ -238,3 +243,33 @@ interface IOUnsafeRun : UnsafeRun<ForIO> {
   override suspend fun <A> unsafe.runNonBlocking(fa: () -> Kind<ForIO, A>, cb: (Either<Throwable, A>) -> Unit) =
     fa().fix().unsafeRunAsync(cb)
 }
+
+@extension
+interface IODispatchers : Dispatchers<ForIO> {
+  override fun default(): CoroutineContext =
+    IODispatchers.CommonPool
```

Comment:
This forces effects-extensions to be JVM-only! I would not go ahead with this refactor as it is.